In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../get_data/turnstile/all_df_cleaned.csv')

In [3]:
df = df.loc[df['division'] == 'IRT']

In [4]:
df.drop('Unnamed: 0',axis=1,inplace=True)

In [5]:
df

control_area  unit       scp          station train_lines division  \
599              R101  R001  02-00-00      SOUTH FERRY          1R      IRT   
600              R101  R001  02-00-00      SOUTH FERRY          1R      IRT   
601              R101  R001  02-00-00      SOUTH FERRY          1R      IRT   
602              R101  R001  02-00-00      SOUTH FERRY          1R      IRT   
603              R101  R001  02-00-00      SOUTH FERRY          1R      IRT   
...               ...   ...       ...              ...         ...      ...   
72804707         R730  R431  00-00-04  EASTCHSTER/DYRE           5      IRT   
72804708         R730  R431  00-00-04  EASTCHSTER/DYRE           5      IRT   
72804709         R730  R431  00-00-04  EASTCHSTER/DYRE           5      IRT   
72804710         R730  R431  00-00-04  EASTCHSTER/DYRE           5      IRT   
72804711         R730  R431  00-00-04  EASTCHSTER/DYRE           5      IRT   

                date      time     desc     entries       exits  \
599       10/11/2014   1:00:00  REGULAR     1873724     3709601   
600       10/11/2014   5:00:00  REGULAR     1873725     3709635   
601       10/11/2014   9:00:00  REGULAR     1873792     3709677   
602       10/11/2014  13:00:00  REGULAR     1873960     3709742   
603       10/11/2014  17:00:00  REGULAR     1874175     3709832   
...              ...       ...      ...         ...         ...   
72804707   10/8/2021   5:00:00  REGULAR  1559793580  1728258395   
72804708   10/8/2021   9:00:00  REGULAR  1559793463  1728258434   
72804709   10/8/2021  13:00:00  REGULAR  1559793445  1728258446   
72804710   10/8/2021  17:00:00  REGULAR  1559793395  1728258511   
72804711   10/8/2021  21:00:00  REGULAR  1559793360  1728258559   

                     datetime  
599       2014-10-11 01:00:00  
600       2014-10-11 05:00:00  
601       2014-10-11 09:00:00  
602       2014-10-11 13:00:00  
603       2014-10-11 17:00:00  
...                       ...  
72804707  2021-10-08 05:00:00  
72804708  2021-10-08 09:00:00  
72804709  2021-10-08 13:00:00  
72804710  2021-10-08 17:00:00  
72804711  2021-10-08 21:00:00  

[26436214 rows x 12 columns]

In [6]:
df['train_lines'].value_counts()

6               4233468
1               3775693
7               3290417
25              2383920
23              1382161
123             1241094
4567S           1064038
45               919398
4                833061
3                642037
2345ACJZ         579848
456              563819
2345BDNQR        353325
456NQRW          347468
123ACE           342472
2345R            339035
1237ACENQRSW     335761
4BD              302702
1237ACENQRS      287311
5                281804
34               278548
456JZ            260684
BDNQR2345        244963
123FLM           230490
1RW              224470
1ABCD            215044
7BDFM            153816
2345             150356
456NQR           142842
456LNQRW         141669
6DF              139514
2345S            123238
245              106058
1AC               98439
7EFMR             76472
2                 76303
1R                65943
7NQW              65726
23ACE             61379
456LNQR           54839
7NQ               26589
Name: train_line

df_7nq = df.loc[df['train_lines']=='7NQ']
df_7nq.to_csv('../turnstile/7NQ.csv')

df_456lnqr = df.loc[df['train_lines']=='456LNQR']
df_456lnqr.to_csv('../turnstile/456LNQR.csv')

df_23ace = df.loc[df['train_lines']=='23ACE']
df_23ace.to_csv('../turnstile/23ACE.csv')

df_7nqw = df.loc[df['train_lines']=='7NQW']
df_7nqw.to_csv('../turnstile/7NQW.csv')

df_1r = df.loc[df['train_lines']=='1R']
df_1r.to_csv('../turnstile/1R.csv')

df_2 = df.loc[df['train_lines']=='2']
df_2.to_csv('../turnstile/2.csv')

df_7efmr = df.loc[df['train_lines']=='7EFMR']
df_7efmr.to_csv('../turnstile/456lnqr.csv')

df_1ac = df.loc[df['train_lines']=='1AC']
df_1ac.to_csv('../turnstile/1AC.csv')

df_245 = df.loc[df['train_lines']=='245']
df_245.to_csv('../turnstile/245.csv')

df_2345S = df.loc[df['train_lines']=='2345S']
df_2345S.to_csv('../turnstile/2345S.csv')

df1 = df.loc[df['train_lines']=='6DF']
df.to_csv('../turnstile/6DF.csv')

In [7]:
df['train_lines'].value_counts()

6               4233468
1               3775693
7               3290417
25              2383920
23              1382161
123             1241094
4567S           1064038
45               919398
4                833061
3                642037
2345ACJZ         579848
456              563819
2345BDNQR        353325
456NQRW          347468
123ACE           342472
2345R            339035
1237ACENQRSW     335761
4BD              302702
1237ACENQRS      287311
5                281804
34               278548
456JZ            260684
BDNQR2345        244963
123FLM           230490
1RW              224470
1ABCD            215044
7BDFM            153816
2345             150356
456NQR           142842
456LNQRW         141669
6DF              139514
2345S            123238
245              106058
1AC               98439
7EFMR             76472
2                 76303
1R                65943
7NQW              65726
23ACE             61379
456LNQR           54839
7NQ               26589
Name: train_line

In [13]:
df1 = df.loc[df['train_lines']=='25']
df_first = df1[:800000]
df_second = df1[800000:1600000]
df_third = df1[1600000:]
# df_fourth = df1[2400000:]
# df_fifth = df1[3200000:3800000]
# df_sixth = df1[3800000:]
df_first.to_csv('../turnstile/25_first.csv')
df_second.to_csv('../turnstile/25_second.csv')
df_third.to_csv('../turnstile/25_third.csv')
# df_fourth.to_csv('../turnstile/25_fourth.csv')
# df_fifth.to_csv('../turnstile/6_fifth.csv')
# df_sixth.to_csv('../turnstile/6_sixth.csv')

In [11]:
trainz = ['45','4','3','2345ACJZ','456','2345BDNQR','456NQRW','123ACE','2345R','1237ACENQRSW',
          '4BD','1237ACENQRS','5','34','456JZ','BDNQR2345']

In [13]:
for line in trainz:
    df1 = df.loc[df['train_lines']==line]
    df1.to_csv('../turnstile/{}.csv'.format(line))

In [12]:
df1 = df.loc[df['train_lines']=='456LNQRW']
df1.to_csv('../turnstile/456LNQRW.csv')

df1 = df.loc[df['train_lines']=='456NQR']
df1.to_csv('../turnstile/456NQR.csv')

df1 = df.loc[df['train_lines']=='2345']
df1.to_csv('../turnstile/2345.csv')

df1 = df.loc[df['train_lines']=='1ABCD']
df1.to_csv('../turnstile/1ABCD.csv')

df1 = df.loc[df['train_lines']=='1RW']
df1.to_csv('../turnstile/1RW.csv')

df1 = df.loc[df['train_lines']=='123FLM']
df1.to_csv('../turnstile/123FLM.csv')